src/core/CMakeLists.txt

```diff
@@ -239,3 +239,7 @@ endif()
 if(OT_RCP)
     include(radio.cmake)
 endif()
+
+set_property(SOURCE api/instance_api.cpp
+    APPEND PROPERTY COMPILE_DEFINITIONS "PACKAGE_VERSION=\"${OT_PACKAGE_VERSION}\""
+)
```

Comment:
Even though this can reduce the build time, but it seems not scalable having to add `PACKAGE_VERSION` for each source file separately?

Reply:
Not each source file, but only those need this definition. I don't think this macro would be widely used in this project.

Reply:
Not sure if it works, but can we only use PACKAGE_VERSION in one header file and let other sources include this header file?

Reply:
I think it more friendly to @zhanglongxia if generating a header file.

Reply:
Header file may not work since header files are not compile units. 

I think it will work if we add a cpp source file that contains code like:
```
const char * const gPackageVersion = PACKAGE_VERSION;
```
and another header file that contains code like:
```
extern const char * const gPackageVersion; 
```

Then, we can just use `gPackageVersion` in other source files.

Reply:
That being said, I am fine with this PR as is.